bftengine/src/bcstatetransfer/SourceSelector.hpp

```diff
@@ -134,6 +144,17 @@ class SourceSelector {
   uint16_t nominatedPrimaryCounter_ = 0;
   uint16_t minPrePrepareMsgsForPrimaryAwarness_ = 10;
   logging::Logger &logger_;
+
+ public:
+  // Metric
+  concordMetrics::Component metrics_component_;
+  struct Metrics {
+    StatusHandle preferred_replicas_;
+    GaugeHandle current_source_replica_;
+    CounterHandle periodic_source_replacement_;
+    CounterHandle source_same_as_primary_;
+  };
+  mutable Metrics metrics_;
```

Comment:
nit:
why mutable? if you remove - do you get any error from some const function?
check again if you decide to move all metrics inside SS - I don't thin you will need it mutable - but maybe i'm wrong.

Reply:
Thanks but It's required.